AssemblyUnhollower/MetadataAccess/CecilMetadataAccess.cs

```diff
@@ -13,10 +14,19 @@ public class CecilMetadataAccess : IIl2CppMetadataAccess
         public CecilMetadataAccess(IEnumerable<string> assemblyPaths)
         {
             var metadataResolver = new MetadataResolver(myAssemblyResolver);
-            
-            foreach (var sourceAssemblyPath in assemblyPaths)
+
+            Load(assemblyPaths.Select(path => AssemblyDefinition.ReadAssembly(path, new ReaderParameters(ReadingMode.Deferred) { MetadataResolver = metadataResolver })));
+        }
+
+        public CecilMetadataAccess(IEnumerable<AssemblyDefinition> assemblies)
+        {
+            Load(assemblies);
+        }
+
+        private void Load(IEnumerable<AssemblyDefinition> assemblies)
+        {
+            foreach (var sourceAssembly in assemblies)
             {
-                var sourceAssembly = AssemblyDefinition.ReadAssembly(sourceAssemblyPath, new ReaderParameters(ReadingMode.Deferred) {MetadataResolver = metadataResolver});
                 myAssemblyResolver.Register(sourceAssembly);
```

Comment:
Could you comment out/remove use of `myAssemblyResolver` and add a short comment that `assemblies` list is assumed to have assemblies that already share an assembly resolver?

Mainly so that this is documented at least somehow. Il2CppDumper already registers all its ADs under the same resolver which is why not using Il2CppUnhollower's `myAssemblyResolver` does not likely break anything. This however may not be so with other tools that we might use in the future, so it's important to remember this strong assumption.

Reply:
Actually, never mind that. I'll add the comment myself since I need to rebuild this for NuGet anyway

Reply:
`myAssemblyResolver` is still used for unity dlls